src/Morphir/IR/NodeId.elm

```diff
@@ -1,34 +1,60 @@
 module Morphir.IR.NodeId exposing (..)
 
 import Morphir.IR.FQName as FQName exposing (FQName)
+import Morphir.IR.Module exposing (ModuleName)
+import Morphir.IR.Name as Name
+import Morphir.IR.Path as Path exposing (Path)
 
 
 type NodeID
     = TypeID FQName
     | ValueID FQName
+    | ModuleID ModuleName
 
 
 nodeIdFromString : String -> Result String NodeID
 nodeIdFromString str =
-    case String.split ":" str of
-        nodeType :: packageName :: moduleName :: localName :: [] ->
-            let
-                fqn : FQName
-                fqn =
-                    FQName.fqn packageName moduleName localName
-            in
-            case nodeType of
+    case String.split ":" str |> List.head of
+        Just prefix ->
+            case prefix of
+                "Value" ->
+                    let
+                        splitNodeString =
+                            String.split ":" str |> List.drop 1
+                    in
+                    case splitNodeString of
+                        packageName :: moduleName :: localName :: [] ->
+                            Ok (ValueID (FQName.fqn packageName moduleName localName))
+
+                        _ ->
+                            Err <| "Value Not Valid"
+
                 "Type" ->
-                    Ok <| TypeID fqn
+                    let
+                        splitNodeString =
+                            String.split ":" str |> List.drop 1
+                    in
+                    case splitNodeString of
+                        packageName :: moduleName :: localName :: [] ->
+                            Ok (TypeID (FQName.fqn packageName moduleName localName))
 
-                "Value" ->
-                    Ok <| ValueID fqn
+                        _ ->
+                            Err <| "Type Not Valid"
+
+                "Module" ->
+                    let
+                        splitNodeString =
+                            String.split ":" str |> List.drop 1 |> List.map Name.fromString
+                    in
+                    case splitNodeString of
+                        moduleName ->
+                            Ok (ModuleID (Path.fromList moduleName))
 
                 _ ->
-                    Err <| "Unknown Node type: " ++ nodeType
+                    Err <| "Invalid NodeId: " ++ str
 
-        _ ->
-            Err <| "Invalid NodeId: " ++ str
+        Nothing ->
+            Err <| "Empty prefix"
```

Comment:
The message here is misleading. This will happen when the list is empty after the split which would mean that the whole node id is empty.